File: src/main/java/es/uji/sape/model/Student.java
package es.uji.sape.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import static es.uji.sape.model.Itinerary.SOFTWARE_ENGINEERING;
import static es.uji.sape.model.Month.JANUARY;

@Data
@NoArgsConstructor
public final class Student {
    public static final int CODE_MAX_LENGTH = 20;
    public static final int NAME_MAX_LENGTH = 40;

    private @NotNull String code = "";
    private @NotNull String name = "";
    private @NotNull String surname = "";
    private @NotNull Itinerary itinerary = SOFTWARE_ENGINEERING;
    private int passedCredits;
    private float averageScore;
    private int pendingSubjects;
    private @NotNull Month internshipStartSemester = JANUARY;
}

File: src/main/kotlin/es/uji/sape/model/Student.kt
package es.uji.sape.model

import es.uji.sape.model.Itinerary.SOFTWARE_ENGINEERING
import es.uji.sape.model.Month.JANUARY
import java.math.BigDecimal

data class Student(
    var code: String = "",
    var name: String = "",
    var surname: String = "",
    var itinerary: Itinerary = SOFTWARE_ENGINEERING,
    var passedCredits: Int = 0,
    var averageScore: BigDecimal = BigDecimal.ZERO,
    var pendingSubjects: Int = 0,
    var internshipStartMonth: Month = JANUARY
)

